Dependencies/xr/Source/ARKit/XR.mm

```diff
@@ -621,16 +635,17 @@ void DrawFrame() {
 
                     // Finalize rendering here & push the command buffer to the GPU.
                     [commandBuffer commit];
-                    [commandBuffer waitUntilCompleted];
                 }
-                @finally {
+                @catch (NSException* exception) {
                     if (cameraTextureY != nil) {
                         [cameraTextureY setPurgeableState:MTLPurgeableStateEmpty];
                     }
                     
                     if (cameraTextureCbCr != nil) {
                         [cameraTextureCbCr setPurgeableState:MTLPurgeableStateEmpty];
                     }
+                    
+                    [exception raise];
```

Comment:
Does this blow away the original stack trace (this is common in other languages)? Is replacing this with `@throw` the right way to re-throw an exception while maintaining the original stack trace?